### PR TITLE
relationship_descriptor -> name, RelationshipTarget -> HolonCollection

### PR DESCRIPTION
**Pull request by dauphin3**
for issue 105, staged_reference.rs line 81
got me thinking, before proceeding, is an enum for HolonCollection for sure what we want?
if so, what implications are that for the logic in add_related_holons  ?

in the new architecture, could a relationship name have both an associated staged and smart collection?

if not, then it looks like this function and ensure_editable_collection would take some significant rewiring. I wanted to checkin before continuing.

### CODE REVIEW

**Comment by dauphin3** [comment]
fixed in 110

**Comment by evomimic** [comment]
This PR represents a LOT of progress in important areas of the MAP and, for the most part works well. 

However, visual inspection of the sweetest results for the final get_all_holons dance shows an empty relationship map for the Book holon. The preceding commit response showed that holon having two entries in its relationship_map (as expected). This suggests a problem either on SAVING the smartlinks OR on building the relationship map FROM the smartlinks. From the trace output on the get_all_holons dance, it looks like holochain is retrieving links. So the most likely source of the error is on the build relationship map side instead of the save_links side.

Even with debug level tracing enabled, I'm not seeing the expected debug messages from get_relationship_name_from_smartlink:
 _debug!("got: {:?}\n link_tag from smartlink ", link_tag.clone());
 debug!("got {:?}\n relationship_name from link_tag", name.clone());_

This suggests the function is not being called. I tried doing a full nom run clean and rebuilding everything, but the test results were the same.


It is difficult to verify the result, so the proposal in the issue that we just using visual inspection of test results is probably insufficient. We should enhance debug tracing to provide better visibility into what's happening and we should ALSO enhance our test logic to do automatic verification of test results.

There is also bit of minor re-factoring I'm going to recommended.

**Comment by evomimic** [comment]
ah well, duh!  We don't have the  get_related_holons query dance implemented until issue #107, so there is no way to exercise the functionality we've implemented here until we have that dance and can add a test step for it to the fixture. Guess that's why I said we couldn't really test this in 105.

Please ignore my comments about test results.

I want to try a bit of re-factoring on a few functions before merging this PR. But, otherwise, everything looks pretty good.

File: crates/coordinator/holons/src/smart_collection.rs
[the remark this answers is not in the document]
Comment (evomimic) [reply crates/coordinator/holons/src/smart_collection.rs:1]: removed in [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884)

File: crates/coordinator/holons/src/staged_collection.rs
[the remark this answers is not in the document]
Comment (evomimic) [reply crates/coordinator/holons/src/staged_collection.rs:1]: Removed in [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884)

File: crates/coordinator/holons/src/holon_collection.rs
[the remark this answers is not in the document]
Comment (evomimic) [reply crates/coordinator/holons/src/holon_collection.rs:1]: Re-implemented in [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884)

[the remark this answers is not in the document]
Comment (evomimic) [reply crates/coordinator/holons/src/holon_collection.rs:1]: Changed in [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884)

File: crates/coordinator/holons/src/relationship.rs
[the remark this answers is not in the document]
Comment (evomimic) [reply crates/coordinator/holons/src/relationship.rs:1]: Items 1 and 3 resolved in [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884).

**Review by evomimic — APPROVED** [review]
Commit [minor refactoring and cleanup, test and sweetest pass](https://github.com/evomimic/map-holons/pull/108/commits/79776a06143348fdf387c586cf84c421ca62c884) resolved all the problems I identified during the review  EXCEPT for one problem that will be resolved in  Issue https://github.com/evomimic/map-holons/issues/114.

Confirmed all sweetest cases and tryorama tests pass.

**Pull request merged**